### ISSUE TITLE
[Python]Report ' RuntimeWarning: invalid value encountered in double_scalars' when run cv.

### ISSUE THREAD

**Issue by zyoohv** [Microsoft/LightGBM#1366]
## Environment info
Operating System: Ubuntu 16.04
Python version:3.5

## Error Message:

[517]   cv_agg's auc: 0.706559 + 0.00383309
[518]   cv_agg's auc: 0.706564 + 0.00382547
[519]   cv_agg's auc: 0.70659 + 0.00380338
[520]   cv_agg's auc: 0.706591 + 0.00382418
[521]   cv_agg's auc: 0.706558 + 0.00382061
[522]   cv_agg's auc: 0.70654 + 0.00382207
[523]   cv_agg's auc: 0.706538 + 0.00382968
[524]   cv_agg's auc: 0.706569 + 0.00383165
[525]   cv_agg's auc: 0.706619 + 0.00379484
[526]   cv_agg's auc: 0.706627 + 0.00376438
[527]   cv_agg's auc: 0.706621 + 0.00379075
[528]   cv_agg's auc: 0.706603 + 0.00376594
[529]   cv_agg's auc: 0.70661 + 0.00377383
[530]   cv_agg's auc: 0.706594 + 0.00379105
/usr/local/lib/python3.5/dist-packages/numpy/core/fromnumeric.py:2957: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
/usr/local/lib/python3.5/dist-packages/numpy/core/_methods.py:80: RuntimeWarning: invalid value encountered in double_scalar
  ret = ret.dtype.type(ret / rcount)

## My source code

#! /usr/bin/python3

import numpy as np
import lightgbm as lgb
import pandas as pd
from setting import *
import scipy

params = \
{
    'task': 'train',
    'boosting_type': 'gbdt',
    'objective': 'binary',
    'metric': {'auc'},
    'train_metric': True,
    'num_leaves': 31,
    # 'min_data_in_leaf': 100,
    'min_child_weight': 50,
    'learning_rate': 0.05,
    'feature_fraction': 0.6,
    'bagging_fraction': 0.7,
    'bagging_freq': 1,
    'verbose': 1,
    'early_stopping_round': 50,
    # 'is_unbalance': True,
    'num_threads': 25
}

train_fea = scipy.sparse.load_npz(root_path + 'train_{}.npz'.format(i))
train_lab = pd.read_csv(root_path + 'label_{}.csv'.format(i)).loc[:, 'label'].values

lgb_train = lgb.Dataset(train_fea, label=train_lab)

lgb.cv(params, lgb_train, verbose_eval=True, num_boost_round=num_iterations, early_stopping_rounds=early_stopping_round)

---

I really doubt that why it happen in [530]?
and what's wrong with me?

**Comment by bbennett36** [comment]
This looks like a problem with your data/numpy, not LightGBM.  
  
This might help - 
https://stackoverflow.com/questions/3767409/invalid-value-encountered-in-double-scalars-warning-possibly-numpy  
  
Yeah it looks like theres something going on with a NaN/null value  -  
`/usr/local/lib/python3.5/dist-packages/numpy/core/fromnumeric.py:2957: RuntimeWarning: Mean of empty slice.`
  
Try this answer if the other one doesn't work -  
https://stackoverflow.com/questions/29688168/mean-nanmean-and-warning-mean-of-empty-slice

**Comment by StrikerRUS** [comment]
Thank you @bbennett36 !
@zyoohv Have you managed to solve your problem?

**Comment by StrikerRUS** [comment]
ping @zyoohv

**Comment by zyoohv** [comment]
Thank you. I have solved the trouble by rewriting my source code.

**Comment by rashibudati** [comment]
@zyoohv , by re-writing do you mean by doing some changes to your code? or re-writng what was actually present earlier in the code?
Thanks in advance
